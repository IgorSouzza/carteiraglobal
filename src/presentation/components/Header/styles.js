import { tc } from '../../utils'

export const wrapper = tc(`
  flex
  items-end
`)

export const image = tc(`
  font-semibold
  text-base
  ml-3.5
`)
