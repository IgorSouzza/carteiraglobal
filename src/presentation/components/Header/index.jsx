import * as S from './styles'

export default function Header () {
  return (
    <div className={S.wrapper}>
      <img src="img/logo.png" alt="Logo Planejador de Poupança" />
      <h1 className={S.image}>Planejador de Poupança</h1>
    </div>
  )
}
