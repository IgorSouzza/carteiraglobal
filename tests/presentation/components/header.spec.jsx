import { render, screen } from '@testing-library/react'
import Header from '../../../src/presentation/components/Header'

describe('<Header />', () => {
  it('should render correctly', () => {
    const { container } = render(<Header />)

    const logo = screen.getByRole('img', { name: /planejador de poupança/i })
    const text = screen.getByRole('heading', { name: /planejador de poupança/i })

    expect(logo).toBeInTheDocument()
    expect(text).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
