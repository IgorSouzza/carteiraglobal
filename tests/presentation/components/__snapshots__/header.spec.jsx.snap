// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<div
  class="flex items-end"
>
  <img
    alt="Logo Planejador de Poupança"
    src="img/logo.png"
  />
  <h1
    class="font-semibold text-base ml-3.5"
  >
    Planejador de Poupança
  </h1>
</div>
`;
